File: python/valid_mountain_array.py
from typing import List


class Solution:
    def validMountainArray(self, numbers: List[int]) -> bool:
        isIncreasing = True
        index = 1
        while index < len(numbers):
            if numbers[index] == numbers[index - 1]: return False
            if numbers[index] < numbers[index - 1] and isIncreasing:
                if index == 1: return False
                isIncreasing = False
            if numbers[index] > numbers[index - 1] and not isIncreasing:
                return False
            index += 1
        return not isIncreasing

File: src/ValidMountainArray.java
public class ValidMountainArray {
    public boolean validMountainArray(int[] arr) {
        boolean isIncreasing = true;
        for (int index = 1 ; index < arr.length ; index++) {
            if (arr[index] == arr[index - 1]) return false;
            if (arr[index] < arr[index - 1] && isIncreasing) {
                if (index == 1) return false;
                isIncreasing = false;
            } else if (arr[index] > arr[index - 1] && !isIncreasing) {
                return false;
            }
        }
        return !isIncreasing;
    }
}
